cupy/manipulation/add_remove.py

```diff
@@ -13,7 +13,63 @@
 # TODO(okuta): Implement resize
 
 
-# TODO(okuta): Implement trim_zeros
+_first_non_zero_krnl = ReductionKernel(
+    'raw S data, T indices, T len, int64 mode',
+    'T y',
+    '''mode? data[indices] ? indices : len
+           : data[len - 1 - indices] ? len - 1 - indices : T(-1)''',
```

Comment:
How about supporting only for `mode=0`, and calling `len - _first_nonzero_kernel(filt[::-1], ...) - 1` to get the value of `end`?

Reply:
I thought filt[::-1] will launch kernels multiple times and that is why I changed it to indexing inside the kernel.